samples/nrf9160/gps/src/supl_support.c

```diff
@@ -0,0 +1,154 @@
+#include <zephyr.h>
+#include <supl_session.h>
+#include <stdio.h>
+#include <logging/log.h>
+
+#define SUPL_SERVER "supl.google.com"
+
+static int            supl_fd;
+
+int open_supl_socket(void)
+{
+	int err;
+	int proto;
+	u16_t port;
+	struct nrf_addrinfo *addr;
+	struct nrf_addrinfo *info;
+
+	proto = NRF_IPPROTO_TCP;
+	port = nrf_htons(7276);
+
+	struct nrf_addrinfo hints = {
+		.ai_family = NRF_AF_INET,
+		.ai_socktype = NRF_SOCK_STREAM,
+		.ai_protocol = proto,
+		/* Either a valid,
+		 * NULL-terminated access point name or NULL.
+		 */
+		.ai_canonname = NULL,
+	};
+
+	err = nrf_getaddrinfo(SUPL_SERVER, NULL, &hints, &info);
+	if (err) {
+		printk("Failed to resolve hostname %s on IPv4, errno: %d)\n",
+			log_strdup(SUPL_SERVER),
+			errno);
+		return -1;
+	}
+
+	supl_fd = nrf_socket(NRF_AF_INET, NRF_SOCK_STREAM, proto);
```

Comment:
We do plain TCP/IP here which is fully offloaded, I suggest to use the posix API then.

Reply:
TCP code updated to use offloaded socket API